src/polyswarm_api/__init__.py

```diff
@@ -735,58 +736,110 @@ def _reveal_closed(self, result):
+            params['with_bounty_results'] = 'true'
+
+        if with_metadata:
+            params['with_metadata'] = 'true'
+
         async with self.get_semaphore:
             logger.debug('Reading results with api-key %s', self.api_key)
             async with aiohttp.ClientSession() as session:
                 try:
-                    async with session.get('{hunt_uri}/{scan_type}/results'.format(hunt_uri=self.hunt_uri,
+                    async def _make_request(hunt_uri, params):
+                        async with session.get('{hunt_uri}/{scan_type}/results'.format(hunt_uri=hunt_uri,
                                                                                    scan_type=scan_type),
                                            params=params,
                                            headers={'Authorization': self.api_key}) as raw_response:
-                        try:
-                            response = await raw_response.json()
-                        except Exception:
-                            response = await raw_response.read() if raw_response else 'None'
-                            logger.error('Received non-json response from PolySwarm API: %s', response)
-                            response = {'status': 'error', 'result': []}
-                        if raw_response.status // 100 != 2:
-                            errors = response.get('errors')
-                            logger.error('Error reading from PolySwarm API: %s', errors)
-                            response = {'status': 'error', 'result': []}
-                        return response
-                except Exception:
+
+
+                            try:
+                                response = await raw_response.json()
+                            except Exception:
+                                response = await raw_response.read() if raw_response else 'None'
+                                logger.error('Received non-json response from PolySwarm API: %s', response)
+                                response = {'status': 'error', 'result': []}
+                            if raw_response.status // 100 != 2:
+                                errors = response.get('errors')
+                                logger.error('Error reading from PolySwarm API: %s', errors)
+                                response = {'status': 'error', 'result': []}
+                            return response
+
+                    agg_response = await _make_request(self.hunt_uri, params)
+
+                    if all_results and not agg_response['status'] == 'error':
+                        total_results = int(agg_response['result']['total'])
+                        for offset in range(limit, total_results, limit):
+                            params['offset'] = offset
+                            response = await _make_request(self.hunt_uri, params)
+                            agg_response['result']['results'].extend(response['result']['results'])
```

Comment:
Would be nice, at some point, to be able to yield results and output them to the user as they come in instead of waiting for all requests to finish. One way to do it is to pass the formatter to the loop and let it make multiple calls. Another way is for the requester to be a generator that yields values as the formatter iterates over it. Either way, it is just an idea for the future.

Reply:
That's a great idea, I'll keep that in mind when we get to the Big Refactor, hopefully soon.